external/VideoStreamer/src/VideoStreamer.cpp

```diff
@@ -0,0 +1,316 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+#include "VideoStreamerInternal.h"
+#include <iostream>
+#pragma comment(lib, "avformat.lib")
+using namespace winrt;
+//#define TIGHT_LATENCY_CONTROL
+#define VERBOSITY  0
+
+#ifdef TIGHT_LATENCY_CONTROL
+    #define MAX_SINK_LATENCY 100
+    uint32_t g_dropCount;
+#endif
+
+#if (VERBOSITY > 0)
+    #define INFOLOG1 printf 
+#else 
+#define INFOLOG1() 
+#endif
+#if (VERBOSITY > 1)
+    #define INFOLOG2 printf 
+#else
+   #define INFOLOG2()
+#endif
+
+    struct GUIDComparer
+{
+    bool operator()(const GUID& Left, const GUID& Right) const
+    {
+        // comparison logic goes here
+        return memcmp(&Left, &Right, sizeof(Right)) < 0;
+    }
+};
+std::map <GUID, AVCodecID, GUIDComparer> g_CodecMapMFtoFF =
+{ {MFVideoFormat_H264, AVCodecID::AV_CODEC_ID_H264},
+  {MFVideoFormat_MPEG2, AVCodecID::AV_CODEC_ID_MPEG2VIDEO}
+};
+
+bool VideoStreamerFFmpeg::s_FFmpegInitDone = false;

```

Comment:
>s_FFmpegInitDone [](start = 26, length = 16)

I think this should be assigned in the header

Reply:
It is a static member. It has to be initialized in cpp file to not have duplicate symbols in linker. Or I can use inline in the header which works only for c++17.  What is the potential issue with having the initialzer here?

Reply:
I think you need to use interlocked commands on the static then since you are updating the value

Reply:
There is no harm in calling ffmpeg init functions multiple times for just the corner case of when multiple CreateInstance are called simultaneously.  Interlocks are not really required to protect functionality.